stl/inc/format

```diff
@@ -2859,122 +2862,142 @@ _NODISCARD auto make_wformat_args(const _Args&... _Vals) {
     return _Format_arg_store<wformat_context, _Args...>{_Vals...};
 }
 
-template <class _Out, class _CharT>
-using format_args_t = basic_format_args<basic_format_context<_Out, _CharT>>;
+template <class _CharT, class... _Args>

```

Comment:
I am wondering, do we even need `_Args`?

Reply:
Or better phrased, I believe it would improve throughput, if for the actual compile time arg checking we have a templateized member function that takes `_Args`

That way we would have one type with `N` member functions rather than `N` types with one member function

Reply:
How do you except then to make the checks compile-time? They must be done in the constructor of `_Fmt_basic_string`.

Reply:
Honestly I am wondering whether one could do
```cpp
template <class... _Types>
_NODISCARD size_t formatted_size(const wstring_view _Fmt, const _Types&... _Args) {
_NODISCARD size_t formatted_size(const _Fmt_wstring _Fmt, const _Types&... _Args) {
    static_assert(_Fmt._Is_valid(_Args...));
    _Fmt_counting_buffer<wchar_t> _Buf;
    _STD vformat_to(_Fmt_wit{_Buf}, _Fmt, _STD make_wformat_args(_Args...));
    _STD vformat_to(_Fmt_wit{_Buf}, _Fmt._Str, _STD make_wformat_args(_Args...));
    return _Buf._Count();
}
```

That said It would definitely be a worse error message / user experience

Reply:
No, that wouldn't be valid because only constructor of `_Fmt_string` can be constant evaluated. From the point of the function it's a normal runtime argument.